DB/src/main/java/io/deephaven/db/v2/by/ComboAggregateFactory.java

```diff
@@ -1387,88 +1397,116 @@ public ComboBy getOut() {
         }
 
         @Override
         public void visit(First first) {
-            out = AggFirst(asMatchPair(first));
+            out = Agg(AggType.First, MatchPair.of(first.addition()));
         }
 
         @Override
         public void visit(Last last) {
-            out = AggLast(asMatchPair(last));
+            out = Agg(AggType.Last, MatchPair.of(last.addition()));
         }
 
         @Override
         public void visit(Std std) {
-            out = AggStd(asMatchPair(std));
+            out = Agg(AggType.Std, MatchPair.of(std.addition()));
         }
 
         @Override
         public void visit(Med med) {
-            out = AggPct(0.50d, med.averageMedian(), asMatchPair(med));
+            out = Agg(new PercentileByStateFactoryImpl(0.50d, med.averageMedian()), MatchPair.of(med.addition()));
         }
 
         @Override
         public void visit(Pct pct) {
-            out = AggPct(pct.percentile(), pct.averageMedian(), asMatchPair(pct));
+            out = Agg(new PercentileByStateFactoryImpl(pct.percentile(), pct.averageMedian()), MatchPair.of(pct.addition()));
         }
 
         @Override
         public void visit(WSum wSum) {
-            out = AggWSum(wSum.weight().name(), asMatchPair(wSum));
+            out = Agg(new WeightedSumStateFactoryImpl(wSum.weight().name()), MatchPair.of(wSum.addition()));
         }
 
         @Override
         public void visit(WAvg wAvg) {
-            out = AggWSum(wAvg.weight().name(), asMatchPair(wAvg));
+            out = Agg(new WeightedAverageStateFactoryImpl(wAvg.weight().name()), MatchPair.of(wAvg.addition()));
         }
 
         @Override
         public void visit(Count count) {
-            out = AggCount(count.column().name());
+            out = new CountComboBy(count.column().name());
         }
 
         @Override
         public void visit(CountDistinct countDistinct) {
-            out = AggCountDistinct(countDistinct.countNulls(), asMatchPair(countDistinct));
+            out = Agg(new CountDistinctStateFactory(countDistinct.countNulls()), MatchPair.of(countDistinct.addition()));
         }
 
         @Override
         public void visit(Distinct distinct) {
-            out = AggDistinct(distinct.includeNulls(), asMatchPair(distinct));
+            out = Agg(new DistinctStateFactory(distinct.includeNulls()), MatchPair.of(distinct.addition()));
         }
 
         @Override
         public void visit(Array array) {
-            out = AggArray(asMatchPair(array));
+            out = Agg(AggType.Array, MatchPair.of(array.addition()));
+        }
+
+        @Override
+        public void visit(Unique unique) {
+            out = Agg(new UniqueStateFactory(unique.includeNulls()), MatchPair.of(unique.addition()));
+        }
+
+        @Override
+        public void visit(SortedFirst sortedFirst) {
+            for (SortColumn column : sortedFirst.columns()) {
+                if (column.order() != SortColumn.Order.ASCENDING) {
```

Comment:
I think we should take this out and not offer things we don't support (until we do).

Reply:
So, I've taken another approach here to see what you think - I'm throwing an error at *construction* time instead of at consumption time. (See SortedFirst/SortedLast)